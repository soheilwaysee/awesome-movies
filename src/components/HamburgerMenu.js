import React from "react";
import styles from "./HamburgerMenu.module.css";

const HamburgerMenu = ({ setShowSideBar }) => (
  <button
    data-testid="button"
    type="button"
    onClick={() => setShowSideBar(show => !show)}
    className={styles.wrapper}
  >
    <span className={styles.bar} />
    <span className={styles.bar} />
    <span className={styles.bar} />
  </button>
);

export default HamburgerMenu;
