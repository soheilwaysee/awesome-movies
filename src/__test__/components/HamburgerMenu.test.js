import React from "react";
import { render, cleanup, fireEvent } from "@testing-library/react";
import "jest-dom/extend-expect";
import HamburgerMenu from "../../components/HamburgerMenu";

afterEach(cleanup);

it("renders HamburgerMenu", async () => {
  const setShowSideBar = jest.fn(show => !show);
  const { container, findByTestId } = render(
    <HamburgerMenu setShowSideBar={setShowSideBar} />
  );
  const button = await findByTestId("button");
  fireEvent.click(button);
  expect(setShowSideBar).toHaveBeenCalledTimes(1);
  fireEvent.click(button);
  expect(setShowSideBar).toHaveBeenCalledTimes(2);
  expect(container.firstChild).toMatchSnapshot();
});
