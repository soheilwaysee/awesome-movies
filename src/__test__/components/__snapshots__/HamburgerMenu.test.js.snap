// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders HamburgerMenu 1`] = `
<button
  class="wrapper"
  data-testid="button"
  type="button"
>
  <span
    class="bar"
  />
  <span
    class="bar"
  />
  <span
    class="bar"
  />
</button>
`;
